Ginger/GingerCore/Drivers/JavaDriverLib/JavaDriver.cs

```diff
@@ -1924,24 +2016,35 @@ List<ElementInfo> IWindowExplorer.GetVisibleControls(List<eElementType> filtered
             else
             {
                 List<PayLoad> ControlsPL = Response.GetListPayLoad();
-                foreach (PayLoad pl in ControlsPL)
+                foreach (var pl in ControlsPL)
                 {
+                    if (mStopProcess)
+                    {
+                        break;
+                    }
                     JavaElementInfo ci = (JavaElementInfo)GetControlInfoFromPayLoad(pl);
 
-                    if(isPOMLearn)
+                    if (isPOMLearn)
                     {
-                        ((IWindowExplorer)this).LearnElementInfoDetails(ci);
-                        // set the Flag in case you wish to learn the element or not
-                        bool learnElement = true;
-                        if (filteredElementType != null)
+                        if (ci.ElementType.Contains("browser") && filteredElementType.Contains(eElementType.Browser))
```

Comment:
we can check it using enum instead of contains,  ci.ElementTypeEnum == eElementType.Browser